sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -839,6 +840,20 @@ func withRootFsShared() oci.SpecOpts {
 	}
 }
 
+// withSwapManagement allows the swapon and swapoff syscalls
```

Comment:
I joined a little late to the party:

I think for this function, it should be fine to skip the closure and just use the implementation you suggested in your other comment. The reason being that there aren't any parameters used by the internal function to further customize the configuration created.

This pattern (not using a closure) is something that we already use, a few lines above we call `oci.WithAllDevicesAllowed` which is just a function that doesn't return a closure.

One more thing, could you please squash your commits :sweat_smile: ?

Reply:
👍 I removed the closure.

> One more thing, could you please squash your commits 😅 ?

I can do it if you definitely want me to, but first I just want to make sure that you know that GitHub can squash the commits when you merge? Just select <kbd>Squash and merge</kbd> in the dropdown next to the merge button.

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits

That should give you full control over the commit message and what the final commit looks like.

If that for some reason isn't sufficient then please let me know and I'll be happy to squash it. 👍

Reply:
We are aware of the feature :sweat_smile: , but we favor contributors squashing/fixing up their commits instead of using the tool provided by GitHub. If you are OK with us squashing the commits for you, we can do it!

Reply:
I squashed the commits. 👍